homeassistant/components/cover/zwave.py

```diff
@@ -110,24 +121,43 @@ class ZwaveGarageDoor(zwave.ZWaveDeviceEntity, CoverDevice):
     def __init__(self, values):
         """Initialize the zwave garage door."""
         ZWaveDeviceEntity.__init__(self, values, DOMAIN)
+        self._cover_state_list = None
+        self._cover_state = None
         self.update_properties()
 
     def update_properties(self):
         """Handle data changes for node values."""
-        self._state = self.values.primary.data
+        self._cover_state = ZWAVE_STATES.get(self.values.primary.data)
+        cover_state_list = self.values.primary.data_items
+        if cover_state_list:
+            self._cover_state_list = list(cover_state_list)
```

Comment:
We need to do a little magic here to map the zwave state list to a hass state list:
```python
            self._cover_state_list = \
                [ZWAVE_STATES[item] for item in cover_state_list]
```